clippy_lints/src/methods/manual_is_variant_and.rs

```diff
@@ -1,18 +1,21 @@
 use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::get_parent_expr;
 use clippy_utils::msrvs::{self, Msrv};
-use clippy_utils::source::snippet;
+use clippy_utils::source::{snippet, snippet_opt};
 use clippy_utils::ty::is_type_diagnostic_item;
 use rustc_errors::Applicability;
+use rustc_hir::{BinOpKind, Expr, ExprKind};
 use rustc_lint::LateContext;
-use rustc_span::{Span, sym};
+use rustc_middle::ty::{self, GenericArg};
+use rustc_span::{BytePos, Span, Symbol, sym};
 
 use super::MANUAL_IS_VARIANT_AND;
 
 pub(super) fn check<'tcx>(
     cx: &LateContext<'_>,
-    expr: &'tcx rustc_hir::Expr<'_>,
-    map_recv: &'tcx rustc_hir::Expr<'_>,
-    map_arg: &'tcx rustc_hir::Expr<'_>,
+    expr: &'tcx Expr<'_>,
+    map_recv: &'tcx Expr<'_>,
+    map_arg: &'tcx Expr<'_>,
```

Comment:
I think all those lifetimes can be removed, including the generic `'tcx` parameter.

Reply:
Ah indeed!